service/history/task/cross_cluster_source_task_executor.go

```diff
@@ -22,30 +22,410 @@
+		shard:          shard,
+		executionCache: executionCache,
+		logger:         logger,
+		metricsClient:  shard.GetMetricsClient(),
+		config:         config,
+	}
 }
 
 func (t *crossClusterSourceTaskExecutor) Execute(
 	task Task,
 	shouldProcessTask bool,
 ) error {
-	panic("crossClusterSourceTaskExecutor not implemented")
+	sourceTask, ok := task.(*crossClusterSourceTask)
+	if !ok {
+		return errUnexpectedTask
+	}
+
+	if !shouldProcessTask {
+		// this should not happen for cross cluster task
+		// shouldProcessTask parameter will be deprecated after
+		// 3+DC task lifecycle is done.
+		return errors.New("encounter shouldProcessTask equals to false when processing cross cluster task at source cluster")
+	}
+
+	if err := t.verifyDomainActive(sourceTask); err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), taskDefaultTimeout)
+	defer cancel()
+
+	var err error
+	switch task.GetTaskType() {
+	case persistence.CrossClusterTaskTypeStartChildExecution:
+		err = t.executeStartChildExecutionTask(ctx, sourceTask)
+	case persistence.CrossClusterTaskTypeCancelExecution:
+		err = t.executeCancelExecutionTask(ctx, sourceTask)
+	case persistence.CrossClusterTaskTypeSignalExecution:
+		err = t.executeSignalExecutionTask(ctx, sourceTask)
+	// TODO: implement the execution logic for recordChildCompleted and applyParentClosePolicy task
+	default:
+		err = errUnknownCrossClusterTask
+	}
+
+	if err == nil {
+		t.setTaskState(sourceTask, ctask.TaskStateAcked, processingStateResponseReported)
+	} else if err == errContinueExecution {
+		err = nil
+	}
+
+	return err
+}
+
+func (t *crossClusterSourceTaskExecutor) executeStartChildExecutionTask(
+	ctx context.Context,
+	task *crossClusterSourceTask,
+) (retError error) {
+
+	taskInfo := task.GetInfo().(*persistence.CrossClusterTaskInfo)
+	wfContext, mutableState, release, err := loadWorkflowForCrossClusterTask(ctx, t.executionCache, taskInfo, t.metricsClient, t.logger)
+	if err != nil || mutableState == nil {
+		return err
+	}
+	defer func() { release(retError) }()
+
+	initiatedEventID := taskInfo.ScheduleID
+	childInfo, ok := mutableState.GetChildExecutionInfo(initiatedEventID)
+	if !ok {
+		return nil
+	}
+
+	ok, err = verifyTaskVersion(t.shard, t.logger, taskInfo.DomainID, childInfo.Version, taskInfo.Version, task)
+	if err != nil || !ok {
+		return err
+	}
+
+	initiatedEvent, err := mutableState.GetChildExecutionInitiatedEvent(ctx, initiatedEventID)
+	if err != nil {
+		return err
+	}
+
+	if task.ProcessingState() == processingStateInvalidated {
+		return t.generateNewTask(ctx, wfContext, mutableState, task)
+	}
+
+	// handle common errors
+	failedCause := task.response.FailedCause
+	if failedCause != nil {
+		switch *failedCause {
+		case types.CrossClusterTaskFailedCauseDomainNotActive:
+			return t.generateNewTask(ctx, wfContext, mutableState, task)
+		case types.CrossClusterTaskFailedCauseWorkflowNotExists:
+			return &types.EntityNotExistsError{}
```

Comment:
what about provide some error detail

Reply:
Sure I can add the details here, but I think the entire entity not exists error will simply be ignored in the handleError() method of cross cluster task type as this is an expected error.